src/designer/designer_packages/elsa-workflows-designer/src/modules/workflow-definitions/plugins/workflow-definitions-plugin.tsx

```diff
@@ -126,7 +126,7 @@ export class WorkflowDefinitionsPlugin implements Plugin {
       await this.workflowDefinitionEditorElement.loadWorkflowVersions();
     }
 
-    return definition;
+    return {...definition, consumingWorkflowsBeingUpdated: updatedWorkflow.consumingWorkflowsBeingUpdated};
```

Comment:
Suggestion: `consumingWorkflowsBeingUpdated` -> `affectedWorkflows`. I understand that the process is still happening in the background somewhere, so technically your field name is spot-on, but `affectedWorkflows` feels simpler to interpret mentally and is close enough to the truth.

Reply:
I see what you mean. I didn't notice the context of where this code was living, I assumed it would be it's own API call. But in this context, your previous name is better. But even better than better would be to decouple the auto-updating of consuming workflows from the logic that simply saves the current workflow. In other words, the client app would make two calls:

1. Saving the workflow definition, as per usual.
2. Updating the consuming workflows.

This would also simplify the backend logic because that too now becomes separate.
These operations don't need to succeed or fail as a whole either: if the saving of the workflow succeeds, but the publishing of the consuming workflows fails, for whatever reason, that's OK.

What do you think?

Reply:
Perfect. Let's do the signalr thing separately, but separate out the updating of the consuming workflows in this PR - at least, I don't see a reason to push this first and then changing it immediately, but you'll correct me if I'm wrong 😄